Fundamentals/Setup/Server-Setup/runtime-modes.md

```diff
@@ -0,0 +1,105 @@
+
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "Runtime": {
+                "Mode" : "Development"
+            },
+            "ModelsBuilder":{
+                "ModelsMode": "SourceCodeAuto"
+            }
+        }
+    }
+}
+```
+
+Now ensure your models are generated by running Umbraco and going to Settings > Models Builder > Generate models. You can now remove the following properties from your `csproj` project file to enable compilation of Razor views (which also ensures your views don't contain compilation errors and is a prerequisite for enabling `Production` mode):
+
+```xml
+<RazorCompileOnBuild>false</RazorCompileOnBuild>
+<RazorCompileOnPublish>false</RazorCompileOnPublish>
+<CopyRazorGenerateFilesToPublishDirectory>true</CopyRazorGenerateFilesToPublishDirectory>
+```
+
+Fix any compilation errors you might get after this, e.g. if you accidentally referenced deleted models or properties. Running the application should now still show the rendered content and you're now ready to optionally enable `Production` mode on your production environment.
+
+## Production mode
+
+Use `Production` mode to ensure your production environment is running optimally by disabling development features and validating whether specific settings are configured to their recommended production values.
+
+This mode disables both in-memory ModelsBuilder generation (see `Development` mode) and Razor (cshtml) runtime compilation, requiring you to compile your views at build/publish time and enforces the following settings for optimal performance/security:
+- The application is built/published in Release mode (with JIT optimization enabled), e.g. using `dotnet publish --configuration Release`;
+- `Umbraco:CMS:WebRouting:UmbracoApplicationUrl` is set to a valid URL;
+- `Umbraco:CMS:Global:UseHttps` is enabled;
+- `Umbraco:CMS:RuntimeMinification:CacheBuster` is set to a fixed cache buster like `Version` or `AppDomain`;
+- `Umbraco:CMS:ModelsBuilder:ModelsMode` is set to `Nothing`.
+
+The recommended approach to enable `Production` mode is to update the `appsettings.Production.json` file (or create one) with the following settings:
+
+```json
+{
+  "Umbraco": {
+    "CMS": {
+      "Runtime": {
+        "Mode": "Production"
+      },
+      "Global": {
+        "UseHttps": true
+      },
+      "ModelsBuilder": {
+        "ModelsMode": "Nothing"
+      },
+      "WebRouting": {
+        "UmbracoApplicationUrl": "https://<REPLACE_WITH_YOUR_PRIMARY_DOMAIN>/"
+      }
+    }
+  }
+}
+```
+
+
+Although you can still edit document types and views (if not running from the published output), changes won't be picked up until you've rebuilt your project or republished the application. Also note that models won't be generated by ModelsBuilder (because the mode is set to `Nothing`), requiring your to do all your changes while in `Development` mode.
```

Comment:
```suggestion
Although you can still edit document types and views (if not running from the published output), changes won't be picked up until you've rebuilt your project or republished the application. 

:::note
Models won't be generated by ModelsBuilder (because the mode is set to `Nothing`), requiring you to do all your changes while in `Development` mode.
:::
```